docs/Chef-CLI-Catalog.md

```diff
@@ -0,0 +1,228 @@
+
+```
+  install                 Install cookbooks from a Policyfile and generate a locked cookbook set
+  update                  Updates a Policyfile.lock.json with latest run_list and cookbooks
+  push                    Push a local policy lock to a policy group on the Chef Infra Server
+  push-archive            Push a policy archive to a policy group on the Chef Infra Server
+  show-policy             Show policyfile objects on the Chef Infra Server
+  diff                    Generate an itemized diff of two Policyfile lock documents
+  export                  Export a policy lock as a Chef Infra Zero code repo
+  clean-policy-revisions  Delete unused policy revisions on the Chef Infra Server
+  clean-policy-cookbooks  Delete unused policyfile cookbooks on the Chef Infra Server
+  delete-policy-group     Delete a policy group on the Chef Infra Server
+  delete-policy           Delete all revisions of a policy on the Chef Infra Server
+  undelete                Undo a delete command
+```
+
+The new binary name proposed is `chef-policy` that could initially be the same policyfiles
+ruby code that lives inside the `chef-cli`. The rest of the sub-commands will continue to
+live inside the `chef-cli` binary.
+
+The new `chef-policy` should contain the following sub-commands:
+```
+$ chef-policy install
+$ chef-policy update
+$ chef-policy push
+$ chef-policy push-archive (?)
+$ chef-policy show
+$ chef-policy diff
+$ chef-policy export
+$ chef-policy clean revisions (?)
+$ chef-policy clean cookbooks (?)
+$ chef-policy group delete (?)
+$ chef-policy delete
+$ chef-policy undelete
+```
+These sub-commands map to the previous top level policy commands. As an example, the
+previous command `chef clean-policy-revisions` would map to the new command
+`chef-policy clean revisions`.
+
+Once the new `chef-policy` exists, we need to update the `chef` catalog to point to
+the new binary and add a section called "Relocated Commands".
+
+The new help command of the `chef` catalog should look like:
+```
+$ chef
+Usage:
+    chef -h/--help
+    chef -v/--version
+    chef command [arguments...] [options...]
+
+Available Commands:
+    env                     Prints environment variables used by Chef Workstation
+    gem                     Runs the 'gem' command in context of the embedded Ruby
+    exec                    Runs the command in context of the embedded ruby
+    shell-init              Initialize your shell to use Chef Workstation as your primary Ruby
+    generate                Generate a new repository, cookbook, or other component
+    describe-cookbook       Prints cookbook checksum information used for cookbook identifier
+    policy                  Manage Chef Policyfiles
+
+Relocated Commands:
+    install                 Use now: chef policy install
+    update                  Use now: chef policy update
+    push                    Use now: chef policy push
+    push-archive            Use now: chef policy push-archive
```

Comment:
Going with the CLI UX `noun verb` guidelines I think the following make more sense:

```
chef policy install
chef policy update
chef policy push
chef policy show
chef policy diff
chef policy export
chef policy delete
chef policy undelete
chef policy-archive push
chef policy-cookbooks clean
chef policy-group delete
chef policy-revisions clean
```

Under the covers, `chef policy-group delete` could still invoke `chef-policy group delete` but that would be an internal implementation detail. It would make it easier for us to implement because it would be less changing of existing code. But I think the public contract is more important to cover in this design doc than the internal implementation of our command routing.

Thoughts?

Reply:
From our conversation over zoom we have agreed that these commands look correct with the exception of `chef-policy push-archive` that will be removed and include into the `chef-policy push` command.

Reply:
More context - I convinced myself that switching to a strictly enforced 'noun verb' syntax means we end up with a bunch of unnecessary nouns (`policy-revision`, `policy-cookbooks`, `policy-groups`, etc.) with 1 verb each.

Instead we talked over some of the existing commands and think the PR correctly documents them. The 'noun' in this case is always 'policy' because all these commands are all connected to Policyfiles.

The only tricky one is `chef-policy clean cookbooks` because it deletes cookbooks that are not referenced by a policyfile. Maybe this becomes an alias for something like `chef cookbooks clean`?